cve_bin_tool/checkers/gnutls.py

```diff
@@ -27,10 +27,13 @@ def get_version(lines, filename):
     ).items():
         for check in binary_names:
             if check in os.path.split(filename)[-1]:
-                return {
-                    "is_or_contains": "is",
-                    "modulename": modulename,
-                    "version": regex_find(lines, *regex),
-                }
+                version_info["is_or_contains"] = "is"
+
+                version = regex_find(lines, *regex)
+    if version != "UNKNOWN":
+        version_info["is_or_contains"] = "contains"
+    if "is_or_contains" in version_info:
+        version_info["modulename"] = "gnutls-cli"
```

Comment:
Looks like you're always returning gnutls-cli here when previously it could also return gnutls-serv.  That's going to cause some bugs.  I think you should probably rethink the logic here so it's a little less interleaved and maybe easier for you to see where it's going?  Here's one possiblity in bad pseudocode:

```
if the pathname matches:
   set is_or_contains to 'is'
   set modulename based on what matched
   set version based on regex_find
   return version_info

otherwise...
version = regex_find(..)
if version != "UNKNOWN":
    set is_or_contains
   set modulename  (But which one is it?  Does the regex tell you?)
   set version
   return version_info

and if we didn't find anything, return a blank version_info 
```

You won't wind up runnnig regex_find twice (even though it's in there twice) because you return before teh secnod time it occurs, but if that doesn't appeal to you aesthetically you could move it around to be run once at the beginnign and then used later as needed.